lib/rack/response.rb

```diff
@@ -38,16 +38,28 @@ def header
       headers
     end
 
+    # :call-seq:
+    #   Response.new(body = nil, status = 200, headers = {}) => response
+    #
     # Initialize the response object with the specified body, status
     # and headers.
     #
-    # @param body [nil, #each, #call, #to_str] the response body.
-    # @param status [Integer] the integer status as defined by the
-    # HTTP protocol RFCs.
-    # @param headers [Hash] a hash of key-value header pairs which
-    # conform to the HTTP protocol RFCs.
+    # If the +body+ is nil, construct an empty response object with internal
+    # buffering.
+    # 
+    # If the body responds to +#to_str+, assume it's a string-like object and
+    # construct a buffered response object containing that string.
+    #
+    # Otherwise it is expected +body+ conforms to the normal requiremnets of a
+    # Rack repsonse body, typically implementing one of +#each+ (enumerable
+    # body) or +#call+ (streaming body).
+    #
+    # The +status+ defaults to +200+ which is the "OK" HTTP status code. You
+    # can provide any other valid status code.
     #
-    # Providing a body which responds to #to_str is legacy behaviour.
+    # The +headers+ must be a +Hash+ of key-value header pairs which conform to
```

Comment:
@jeremyevans do I need to write `+Hash+` here?

Reply:
I don't think you need to.  You can use `Hash` or `hash`.  You only need to use `+Hash+` if you want to force it displayed as code.  Either way is fine with me.

Reply:
I tried out different ways and checked how it was formatted, I'm happy with it being formatted as code for now.